test/filtering.test.js

```diff
@@ -107,6 +115,12 @@ describe('filtering', () => {
     grid._filters.forEach((filter) => expect(filter.$.filter.clientHeight).to.be.greaterThan(0));
   });
 
+  it('should not keep references to filters when column is removed', () => {
```

Comment:
Considering the probably bug above, we might want to add a test case where we check that it keeps the filters for columns that were not removed.